crates/paralegal-policy/src/context.rs

```diff
@@ -219,38 +322,69 @@ impl Context {
             .flat_map(|v| v.iter())
     }
 
-    /// Returns an iterator over all sinks marked with `marker` out of the provided `dsts`.
-    pub fn marked_sinks<'a>(
-        &'a self,
-        dsts: impl IntoIterator<Item = &'a DataSink> + 'a,
-        marker: Marker,
-    ) -> impl Iterator<Item = &'a DataSink> + 'a {
-        dsts.into_iter().filter(move |dst| match dst {
-            DataSink::Argument { function, arg_slot } => self
-                .marker_to_ids
-                .get(&marker)
-                .map(|markers| {
-                    markers.iter().any(|(id, refinement)| {
-                        id == &function.function
-                            && (refinement.on_self()
-                                || refinement.on_argument().contains(*arg_slot as u32).unwrap())
+    /// Returns whether the given CtrlNode has the marker applied to it directly or via its type.
+    pub fn has_marker(&self, marker: Marker, node: &CtrlNode) -> bool {
```

Comment:
This code seem unfinished?

Reply:
It seems you've now filled it in anyway 😄

Reply:
Filled in most of it because I realized the marked_sources fn had some of this functionality already. Still have to fill out the rest and add markers from types in a subsequent PR.